torchao/prototype/mx_formats/mx_tensor.py

```diff
@@ -752,6 +761,14 @@ def mx_addmm(func, types, args, kwargs):
     b = args[2]
     return _addmm_mx_dispatch(a, b, func, bias=bias)
 
+@implements([aten.linear.default])
+def mx_linear(func, types, args, kwargs):
```

Comment:
I think the reshape should live here, the `addmm` op is supposed to have 2d inputs